docs/development_suite/company/project-blueprint.md

```diff
@@ -0,0 +1,263 @@
+:::
+
+*mustache.js* is a web template system, which allows you to generate custom templates by replacing all the general information in the web template with your product or organization information.
+
+You could create a Project template to avoid copy/paste in every new Project having the same base configuration.
+
+At Mia-Platform, for example, we create one template to configure a Project exposed through Traefik configuration and one to configure it without expose it. Therefore, for a company that uses this template, creating this type of Project will be a very simple process.
+
+### Template interpolation variables
+
+You can interpolate the template with some Project data. With *mustache.js*, you could iterate through an array, in order to have some configuration iterated for every environment.
+The values you can use during template interpolation are:
+
+For the Project (accessible using `project.${field}`):
+
+* `projectId`;
+* `name`;
+* `configurationGitPath`;
+* `environments`.
+
+Inside environments, you can access to:
+
+* `envId`;
+* `envPrefix`;
+* `cluster` (an object containing `namespace` and `clusterId` string, `kubeContextVariables` object). Here you can find the variable names for the specified environment where to set the cluster variable;
+* `hosts` (an array of objects, with `host` and `isBackoffice` fields).
+
+An example of template for the `.gitlab-ci.yml` file:
+
+```yml
+include:
+  # job template
+  - project: 'platform/pipelines-templates'
+    file: '/deploy/deploy-job.yml'
+    ref: 'master'
+%#project.environments%
+
+%envId%:
+  stage: release
+  extends: .deploy_job
+
+  variables:
+    KUBE_URL: "\${cluster.kubeContextVariables.KUBE_URL}"
+    KUBE_TOKEN: "\${cluster.kubeContextVariables.KUBE_TOKEN}"
+    KUBE_CA_PEM: "\${cluster.kubeContextVariables.KUBE_CA_PEM}"
+    ENVIRONMENT_PREFIX: "%envPrefix%_"
+
+  only:
+    variables:
+      - $ENVIRONMENT_TO_DEPLOY == "%envId%"
+%/project.environments%
+```
+
+All sections between `%#project.environments%` and `%/project.environments%` will be written for `n` times, where `n` is the number of environments. So, inside the environment, you can use the environment specific fields.
+For other possibilities, please check [mustache.js](https://github.com/janl/mustache.js) documentation.
+
+You may want to write a file or a folder for every environment. To enable that, you can write the file name (or folder) in the template as `%envId%`, which will be interpolated for every environment.
+The interpolation data in those files include the environments fields at the first level (as in `mustache.js` sections), with the Project as a key for every environment.
+
+To see an example of Jenkins pipelines definition, take a look at [this](/development_suite/deploy/pipeline-based/configure-jenkins.md#how-to-automatically-create-jenkins-job-on-project-creation) page of our doc.
+
+## Create default services
+
+There are two ways to add services to a template:
```

Comment:
segnerei che è una cosa, quella degli enableServices che è in deprecazione

Reply:
The information is already given to the user (it says "Deprecated")